server/src/handlers/userHandler.ts

```diff
@@ -114,6 +116,26 @@ export class UserHandler {
     });
   };
 
+  public updateCurrentUser: ExpressHandler<UpdateCurrentUserRequest, UpdateCurrentUserResponse> =
```

Comment:
this endpoint allows you to edit other users' username, no?
we need another check that the `existingUser`'s id below is equal to `res.locals.userId`.

Reply:
No, it doesn't. All profile updates target the current authenticated use as I do the updates query against `res.locals.userId`.